vision_agent/lmm/lmm.py

```diff
@@ -6,100 +6,52 @@
 from pathlib import Path
 from typing import Any, Callable, Dict, List, Optional, Union, cast
 
-import requests
 from openai import AzureOpenAI, OpenAI
 
+import vision_agent.tools as T
 from vision_agent.tools.prompts import CHOOSE_PARAMS, SYSTEM_PROMPT
 
 _LOGGER = logging.getLogger(__name__)
 
-_LLAVA_ENDPOINT = "https://svtswgdnleslqcsjvilau4p6u40jwrkn.lambda-url.us-east-2.on.aws"
-
 
 def encode_image(image: Union[str, Path]) -> str:
     with open(image, "rb") as f:
         encoded_image = base64.b64encode(f.read()).decode("utf-8")
     return encoded_image
 
 
+TextOrImage = Union[str, List[Union[str, Path]]]
+MediaChatItem = Dict[str, TextOrImage]
+
+
 class LMM(ABC):
     @abstractmethod
     def generate(
-        self, prompt: str, images: Optional[List[Union[str, Path]]] = None
+        self, prompt: str, media: Optional[List[Union[str, Path]]] = None
     ) -> str:
         pass
 
     @abstractmethod
     def chat(
         self,
-        chat: List[Dict[str, str]],
-        images: Optional[List[Union[str, Path]]] = None,
+        chat: List[MediaChatItem],
     ) -> str:
         pass
 
     @abstractmethod
     def __call__(
         self,
-        input: Union[str, List[Dict[str, str]]],
-        images: Optional[List[Union[str, Path]]] = None,
+        input: Union[str, List[MediaChatItem]],
     ) -> str:
         pass
```

Comment:
love this change as it could potentially unify the function signature between llm and lmm and making the abstraction simpler.
And on this thought, have you thought about unify the `LLM` and `LMM` abstraction into one?
With this change, it seems that we don't need two separate abstractions, right?

Reply:
Good idea! Added this change